aptos-move/block-executor/src/executor.rs

```diff
@@ -299,6 +305,48 @@ where
         }
     }
 
+    fn validate_commit_ready(
+        txn_idx: TxnIndex,
+        versioned_cache: &MVHashMap<T::Key, T::Tag, T::Value, X, T::Identifier>,
+        last_input_output: &TxnLastInputOutput<T, E::Output, E::Error>,
+    ) -> bool {
+        let read_set = last_input_output
+            .read_set(txn_idx)
+            .expect("Read set must be recorded");
```

Comment:
Can we have no read set?

Reply:
nope

Reply:
Why can't we have an empty read set? A program could simply do nothing right?

Reply:
that means you will have empty CapturedReads object which will be in last_input_output

last_input_output cannot be missing CapturedReads for executed transaction, it is inserted at the end of execute() in executor.rs